src/vs/editor/common/controller/cursorWordOperations.ts

```diff
@@ -438,6 +438,91 @@ export class WordOperations {
+		const startIndex1 = position.column - 1; // deleteRight
+		const startIndex2 = position.column - 2; // deleteLeft
+		const firstNonWhitespace = this._findFirstNonWhitespaceChar(lineContent, startIndex1);
+		const lastNonWhitespace = strings.lastNonWhitespaceIndex(lineContent, startIndex2);
+		if ((startIndex1 + 1 < firstNonWhitespace && lastNonWhitespace + 1 < startIndex2) || (startIndex1 + 1 >= firstNonWhitespace && lastNonWhitespace + 1 < startIndex2) || (startIndex1 + 1 <= firstNonWhitespace && lastNonWhitespace + 1 === startIndex2)) {
+			return new Range(position.lineNumber, lastNonWhitespace + 2, position.lineNumber, firstNonWhitespace + 1);
+		}
+		return null;
+	}
+
+
+	public static deleteWordEntire(ctx: DeleteWordContext, wordNavigationType: WordNavigationType): Range | null {
+		const wordSeparators = ctx.wordSeparators;
+		const model = ctx.model;
+		const selection = ctx.selection;
+		const whitespaceHeuristics = ctx.whitespaceHeuristics;
+
+		if (!selection.isEmpty()) {
+			return selection;
+		}
+
+		if (DeleteOperations.isAutoClosingPairDelete(ctx.autoClosingBrackets, ctx.autoClosingQuotes, ctx.autoClosingPairs.autoClosingPairsOpen, ctx.model, [ctx.selection])) {
+			const position = ctx.selection.getPosition();
+			return new Range(position.lineNumber, position.column - 1, position.lineNumber, position.column + 1);
+		}
+
+		const position = new Position(selection.positionLineNumber, selection.positionColumn);
+
+		let lineNumber = position.lineNumber;
+		let column = position.column;
+		let column_end = position.column;
+		let lineContent = model.getLineContent(position.lineNumber);
+
+		if (lineNumber === 1 && column === 1) {
+			// Ignore deleting at beginning of file
+			return null;
+		}
+
+		if (whitespaceHeuristics) {
+			let r = this._deleteWordEntireWhitespace(model, position);
+			if (r) {
+				return r;
+			}
+		}
+
+		let prevWordOnLine = WordOperations._findPreviousWordOnLine(wordSeparators, model, position);
+
+		if (wordNavigationType === WordNavigationType.WordStart) {
+			if (prevWordOnLine) {
+				column = prevWordOnLine.start + 1;
+				column_end = prevWordOnLine.end + 1;
+				const chCode = lineContent.charCodeAt(column_end - 1);
+				if (chCode === CharCode.Space || chCode === CharCode.Tab) {
+					column_end++;
+				}
+			} else {
+				if (column > 1) {
+					column = 1;
+				} else {
+					lineNumber--;
+					column = model.getLineMaxColumn(lineNumber);
+				}
+			}
+		} else {
+			if (prevWordOnLine && column <= prevWordOnLine.end + 1) {
+				prevWordOnLine = WordOperations._findPreviousWordOnLine(wordSeparators, model, new Position(lineNumber, prevWordOnLine.start + 1));
+			}
+			if (prevWordOnLine) {
+				column = prevWordOnLine.end + 1;
+			} else {
+				if (column > 1) {
+					column = 1;
+				} else {
+					lineNumber--;
+					column = model.getLineMaxColumn(lineNumber);
+				}
+			}
+		}
+
+		return new Range(lineNumber, column, position.lineNumber, column_end);
+	}
```

Comment:
This shares and duplicates a lot from the deleteWordLeft function.  Can then be refactored and share a common function?

Reply:
Thank you so much for your advice. I made a new function '_determineDeleteRange' and replaced the common code by it.